.github/workflows/run-nightly-tests.yml

```diff
@@ -0,0 +1,93 @@
+---
+name: run-nightly-tests
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 01 * * *'
+
+jobs:
+  nightly-tests:
+    runs-on: nvme-nvm
+    steps:
+      - name: Output kernel version
+        run: |
+          uname -a
+      - name: Clean up test device
+        run: |
+          #BDEV0 is an environment variable of the self-hosted runner instance
+          #that contains a valid nvme ctrl name which is capable of the nvm
+          #command set.
+          CONTROLLER=$(echo /dev/${BDEV0} | sed 's/n[0-9]*$//')
+          sudo nvme delete-ns $CONTROLLER -n 0xffffffff
+          sudo nvme format $CONTROLLER -n 0xffffffff -l 0 -f
+          SIZE=$(sudo nvme id-ctrl $CONTROLLER | grep tnvmcap | sed 's/^tnvmcap   : //g' | sed 's/,//g' | awk '{print $1/512}')
+          sudo nvme create-ns -s $SIZE -c $SIZE -f 0 -d 0 --csi=0 $CONTROLLER
+          sudo nvme attach-ns $CONTROLLER -n 1 -c 0
+      - uses: actions/checkout@v4
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install --no-install-recommends -y \
+          meson gcc pkg-config git libjson-c-dev libssl-dev libkeyutils-dev \
+          libdbus-1-dev libpython3-dev pipx python3-dev swig xz-utils
```

Comment:
I think we should prebuild these containers, since in the past this step fails way to often due network SNAFUs.

Reply:
This is installing the dependencies directly on the VM (without a container in between) which gets reset everytime a workflow finishes.
I could install those additional packages when spawning the self-hosted runner.
Otherwise, a cleaner option could be to start using a prebuild container in the VM to run those tests on.
What do you think - should I run the tests additionally in the container within the VM? :)

Reply:
I think we can start with this approach and address it later. The install step usually works for a couple of month followed by few days nothing works. That's why I think we are going to need the pre build containers eventually. But for the time being it's okay to go with this. Let's get it working first before going into too many rabbit holes.

Reply:
Sounds good! I will include a commit for this in the next PR to the test infrastructure :)